aten/src/ATen/native/cuda/AveragePool2d.cu

```diff
@@ -167,95 +268,68 @@ void avg_pool2d_out_cuda_template(
     inputHeight, inputWidth,
     outputHeight, outputWidth);
 
-  Tensor input = input_.contiguous();
+  Tensor input = input_.contiguous(memory_format);
 
   output.resize_({nbatch, nInputPlane, outputHeight, outputWidth});
 
+  output.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
```

Comment:
NITPICK: Maybe move this into the switch condition. Memory restride is not needed for contiguous output.

Reply:
Good point, fixed.